src/ColinHDev/CPlot/provider/DataProvider.php

```diff
@@ -1299,4 +1312,222 @@ public function savePlotRate(Plot $plot, PlotRate $plotRate) : Generator {
+                        "sqlite" => "sql" . DIRECTORY_SEPARATOR . "myplot_sqlite.sql"
+                    ]);
+                case 'mysql':
+                    $sqlSettings = $settings["MySQLSettings"];
+                    $myplotDatabase = $myplotDatabase ?? libasynql::create(CPlot::getInstance(), [
+                        "type" => "mysql",
+                        "mysql" => [
+                            "host" => $sqlSettings["Host"],
+                            "user" => $sqlSettings["Username"],
+                            "password" => $sqlSettings["Password"],
+                            "schema" => $sqlSettings["DatabaseName"],
+                            "port" => $sqlSettings["Port"]
+                        ],
+                        "worker-limit" => ResourceManager::getInstance()->getConfig()->getNested("database.worker-limit", 1)
+                    ], [
+                        "mysql" => "sql" . DIRECTORY_SEPARATOR . "myplot_mysql.sql"
+                    ]);
+                    $records = yield from $myplotDatabase->asyncSelect(self::EXPORT_MYPLOT_PLOTS);
+                    $mergeRecords = yield from $myplotDatabase->asyncSelect(self::EXPORT_MYPLOT_MERGES);
+                    break;
+                case 'yaml':
+                    $filename = "plots.yml";
+                case 'json':
+                    $data = new Config(Path::join(Server::getInstance()->getDataPath(), "plugin_data", "MyPlot", "Data", $filename ?? "plots.json"), Config::DETECT);
+                    $records = array_values($data->get("plots"));
+                    $unparsedMergeRecords = $data->get("merges");
+                    $mergeRecords = [];
+                    foreach($unparsedMergeRecords as $origin => $merges) {
+                        $originData = explode(";", $origin);
+                        foreach($merges as $merge) {
+                            $mergeData = explode(";", $merge);
+                            $mergeRecords[] = [
+                                "level" => $originData[0],
+                                "originX" => $originData[1],
+                                "originZ" => $originData[2],
+                                "mergedX" => $mergeData[0],
+                                "mergedZ" => $mergeData[1]
+                            ];
+                        }
+                    }
+                    break;
+                default:
+                    return; // don't import anything due to invalid data provider
+            }
+            foreach($records as $record) {
+                // validate offline player data
+                $XUID = null;
+                $offlineData = Server::getInstance()->getOfflinePlayerData($record["owner"]);
+                if($offlineData !== null) {
+                    $XUID = $offlineData->getString("LastKnownXUID", null);
+
+                    // register filler player data
+                    yield from $this->updatePlayerData(
```

Comment:
I know, `playerData` is a difficult topic with #65 still being unresolved, but is there a reason, you don't store the player's information if only his name is known?

Reply:
Ah, I forgot to move the function call out of the XUID check to support nullability